lib/chunck/irmin_chunck.ml

```diff
@@ -0,0 +1,421 @@
+(**
+
+Chunck Norris Module.
+The module who split your data on same size chuncks.
+
+*)
+
+
+open Lwt
+open Irmin
+
+
+module Log = Log.Make(struct let section = "CHUNCK" end)
+
+module type RAW = Tc.S0 with type t = Cstruct.t
+
+module type AO_MAKER_RAW =
+  functor (K: Hash.S) ->
+  functor (V: RAW) ->
+  AO with type key = K.t and type value = V.t
+
+					  
+
+module CHUNCK_AO (S:AO_MAKER_RAW) (K:Irmin.Hash.S) (V: RAW) = struct
+
+
+    
+    (* Types *)
+    module AO = S(K)(V)
+    type key = AO.key
+    type value = AO.value
+    type t = {
+	db:AO.t;
+	size:int;
+	hash_length: int;
+	nb_indirect: int;
+	data_length:int;
+      }
+
+	       
+    (* Configuration module *)
+(*    module Conf = struct
+	exception Wrong_Param
+		    
+	let chunck_size =
+	  Irmin.Private.Conf.key
+	    ~doc:"Size of chunck"
+	    "size"
+	    (Ir_conf.int)
+	    4096
+
+	let get_ratio_dir r = (fst r)	
+	let get_ratio_indir r = (snd r)
+	let size_ratio r = (get_ratio_dir r) + (get_ratio_indir r)
+
+      end
+ *)   		    		  		     
+
+    (* Chunck Module 
+
+     All functions for manipulate chuncks representation, a chunck is represented as above : 
+      --------------------------
+     | uint8_t type             | 
+     ---------------------------
+     | uint16_t chunck_length   |
+     ---------------------------
+     | uint32_t total_length    |
+     ---------------------------
+     | byte data[data_length]   |
+     ---------------------------
+     Where type define if the chunck contain data or indirection, size represent the data length to 
+     consider, and data field is the payload. 
+     *)
+	       
+    module Chunck = struct 
+	
+	let hash_length = K.length
+			    
+	let l_type = 1
+	let get_chunck_type x = Cstruct.get_uint8 x 0 				      
```

Comment:
every function here are related to chunks (are they are in the `Chunk` module). You don't need to repeat `chunk` in the function name.